libs/domain/cart/coupon/coupon.component.ts

```diff
@@ -0,0 +1,115 @@
+import { CartComponentMixin, CartService } from '@spryker-oryx/cart';
+import { resolve } from '@spryker-oryx/di';
+import { ContentMixin } from '@spryker-oryx/experience';
+import { LocaleService } from '@spryker-oryx/i18n';
+import { NotificationService } from '@spryker-oryx/site';
+import { AlertType } from '@spryker-oryx/ui';
+import { ButtonSize, ButtonType } from '@spryker-oryx/ui/button';
+import { ColorType } from '@spryker-oryx/ui/link';
+import { hydrate } from '@spryker-oryx/utilities';
+import { LitElement, TemplateResult, html } from 'lit';
+import { query, state } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+import { couponStyles } from './coupon.styles';
+
+@hydrate({ event: ['window:load'] })
+export class CouponComponent extends CartComponentMixin(
+  ContentMixin(LitElement)
+) {
+  static styles = couponStyles;
+
+  @state() hasError = false;
+
+  @query('input[name=coupon]') coupon!: HTMLInputElement;
```

Comment:
```suggestion
  @query('input[name=coupon]') coupon?: HTMLInputElement;
```

I prefer to keep it optional, in case the component html is customised, this is not a given.